### PR TITLE
Make project require public

### PR DESCRIPTION
**Pull request by asakusuma**
Addresses https://github.com/ember-cli/ember-cli/issues/6733

cc @stefanpenner

### CODE REVIEW

**Comment by rwjblue** [comment]
I've added a bunch of reviewers here, I'd love for them to review and confirm that they agree...

File: lib/models/project.js [hunk 1]
```diff
@@ -281,7 +281,7 @@ class Project {
   /**
     Calls `require` on a given module.
```
Comment (rwjblue) [inline lib/models/project.js]: Can you flesh this out a bit more?  The docs are pretty scarce here...

Comment (stefanpenner) [inline lib/models/project.js]: good point, lets make sure the docs are more fleshed out before flipping to public.

* describe what this does
* describe when one may want to use this.

**Comment by asakusuma** [comment]
Unsure why appveyor is failing

```
Build started
git clone -q https://github.com/ember-cli/ember-cli.git C:\projects\ember-cli
git fetch -q origin +refs/pull/6737/merge:
git checkout -qf FETCH_HEAD
The item was not found or does not exist.
```

**Comment by kellyselden** [comment]
@asakusuma https://github.com/ember-cli/ember-cli/issues/6735

**Comment by kellyselden** [comment]
@asakusuma Kicked.

**Comment by chadhietala** [comment]
@homu r+

**Pull request merged**